miqi/execution/task_policy.py

```
@@ -0,0 +1,332 @@
+    "send_message": PHASE_EXTERNAL,
+    "spawn": PHASE_EXTERNAL,
+}
+
+
+def phase_for_tool(tool_name: str) -> str | None:
+    """工具 → 阶段（未知工具返回 None——不计阶段，不弹阶段分）。"""
+    return TOOL_PHASE.get(tool_name)
+
+
+def is_mutation_tool(tool_name: str) -> bool:
+    """mutation gate（GPT P0-2）：WRITE/EXEC/EXTERNAL → True（PlanCard 前禁止）。"""
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)